### ISSUE TITLE
large action head of Qwen2.5-VL-PI,

### ISSUE THREAD

**Issue by Yysrc** [starVLA/starVLA#166]
Great work, thanks for your contribution!
When using Qwen2.5-VL-PI, I noticed that its action head is quite large, around 2B parameters. In comparison, the action head of pi_0 is only 300M.
Are there any ways to reduce the number of parameters in the action head (other than reducing num_layers)?

**Comment by JinhuiYE** [comment]
Thanks for your question! Yes, you can reduce the number of parameters in the action head by decreasing the hidden state dimension inside the DiT (Diffusion Transformer). This is feasible, but you'll need to debug to find the optimal parameters. We'd also welcome a PR from you after you figure it out!

**Comment by Yysrc** [comment]
Thank you for your reply, it's really helpful! 
However, I noticed that there is a line of code in the `LayerwiseFM_ActionHeader_v2.py` file you provided in https://github.com/starVLA/starVLA/issues/113: 
`cross_and_self_feature = torch.cat((vl_embs_list[layer_idx], hidden_states), dim=1)`.
This means that the hidden states of DiT and the hidden states of Qwen need to be consistent. If this line is removed, the hidden states of DiT can be modified arbitrarily. May I ask what considerations led to this design choice?

**Comment by JinhuiYE** [comment]
Great question! The concatenation operation in that line doesn't actually require the hidden states to have the same dimension. Let me explain with attention formulas:

## Attention Mechanism Basics

For attention, we compute:
\[ \text{Attention}(Q, K, V) = \text{softmax}\left(\frac{QK^T}{\sqrt{d_k}}\right)V \]

## Key Insight

The only requirement is that **Q and K have matching dimensions** for the dot product. The original hidden state dimensions can be different!

If we have:
- DiT hidden state: \( h_d \in \mathbb{R}^{d_d} \)
- Qwen hidden state: \( h_q \in \mathbb{R}^{d_q} \)

We can project them to the same dimension \(d_k\):

\[ Q = W_q \cdot h_q, \quad W_q \in \mathbb{R}^{d_k \times d_q} \]
\[ K = W_k \cdot h_d, \quad W_k \in \mathbb{R}^{d_k \times d_d} \]
\[ V = W_v \cdot h_d, \quad W_v \in \mathbb{R}^{d_v \times d_d} \]

## Why We Used Concatenation

The line `cross_and_self_feature = torch.cat((vl_embs_list[layer_idx], hidden_states), dim=1)` was just a design choice assuming they have the same dimension for simplicity. **It's not a fundamental requirement** - you can freely modify DiT's hidden states and add projection layers to align dimensions.

So yes, you're absolutely right - you can remove that line and modify DiT's hidden states arbitrarily, as long as you ensure proper projection before attention!

**Comment by Yysrc** [comment]
Understood. Thank you for your patient explanation.
